tests/testthat/test_merModList.R

```diff
@@ -1,113 +1,126 @@
-  zz <- summary(g1)
-  expect_is(zz, "summary.merModList")
-
-})
-
-# Numerical accuracy of merModList print method----
-context("Numerical accuracy of merModList print method")
-
-test_that("print method for merModList works in general case", {
-  # skip_on_cran()
-  data(grouseticks)
-  grouseticks$HEIGHT <- scale(grouseticks$HEIGHT)
-  grouseticks <- merge(grouseticks, grouseticks_agg[, 1:3], by = "BROOD")
-  grouseticks$TICKS_BIN <- ifelse(grouseticks$TICKS >=1, 1, 0)
-  form <- TICKS_BIN ~ HEIGHT +(1 + HEIGHT|BROOD) + (1|YEAR)
-  modDat <- vector(5, mode="list")
-  for(i in 1:length(modDat)){
-    modDat[[i]] <- grouseticks[sample(x=1:nrow(grouseticks), size = nrow(grouseticks),
-                                      replace=FALSE),]
-  }
-
-  g1 <- glmerModList(formula = form,
-                    data = modDat, family = "binomial",
-                    control = glmerControl(optimizer="bobyqa",
-                                           optCtrl=list(maxfun = 1e6)))
-  g1T <- glmer(form, family = "binomial", data = grouseticks,
-               control = glmerControl(optimizer="bobyqa",
-                                      optCtrl=list(maxfun = 1e6)))
-
-  expect_equal(VarCorr(g1)$stddev$BROOD, attr(VarCorr(g1T)$BROOD, "stddev"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$stddev$YEAR, attr(VarCorr(g1T)$YEAR, "stddev"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$correlation$BROOD, attr(VarCorr(g1T)$BROOD, "corre"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$correlation$YEAR, attr(VarCorr(g1T)$YEAR, "corre"),
-               tolerance = 0.0001)
-
-  form <- TICKS_BIN ~ HEIGHT +(1|BROOD)
-  g1 <- glmerModList(formula = form,
-                     data = modDat, family = "binomial",
-                     control = glmerControl(optimizer="bobyqa",
-                                            optCtrl=list(maxfun = 1e6)))
-  g1T <- glmer(form, family = "binomial", data = grouseticks,
-               control = glmerControl(optimizer="bobyqa",
-                                      optCtrl=list(maxfun = 1e6)))
-
-  expect_equal(VarCorr(g1)$stddev$BROOD, attr(VarCorr(g1T)$BROOD, "stddev"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$stddev$YEAR, attr(VarCorr(g1T)$YEAR, "stddev"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$correlation$BROOD, attr(VarCorr(g1T)$BROOD, "corre"),
-               tolerance = 0.0001)
-  expect_equal(VarCorr(g1)$correlation$YEAR, attr(VarCorr(g1T)$YEAR, "corre"),
-               tolerance = 0.0001)
-
-})
-
-#ICC function----
-context("ICC function")
-
-test_that("ICC function works", {
-  ICC1 <- ICC(outcome = "Reaction", group = "Subject", data = sleepstudy)
-  expect_is(ICC1, "numeric")
-  expect_equal(ICC1, 0.3948896, tol = .001)
-})
+# test merModList functions
+
+#Do merModList objects get built and work----
+context("Do merModList objects get built and work")
+
+old_warn <- getOption("warn")
```

Comment:
For my own information, what does this change here do? Has something changed in lme4 that I've fallen behind on?

Reply:
The existing tests were getting convergence warning messages in `devtools::test`. So I added this to suppress warnings and then turn them back on. see https://stackoverflow.com/questions/16194212/